src/deluge/io/midi/midi_follow.cpp

```diff
@@ -480,6 +480,14 @@ void MidiFollow::offerReceivedCCToMelodicInstrument(ModelStackWithTimelineCounte
 	MelodicInstrument* melodicInstrument = (MelodicInstrument*)clip->output;
 	melodicInstrument->receivedCC(modelStackWithTimelineCounter, fromDevice, match, channel, ccNumber, value,
 	                              doingMidiThru);
+
+	//if you're in automation midi clip view and editing the same CC that was just updated
+	//by a learned midi knob, then re-render the pads on the automation editor grid
+	if (clip->output->type == InstrumentType::MIDI_OUT && getRootUI() == &automationInstrumentClipView) {
+		if (((InstrumentClip*)clip)->lastSelectedParamID == ccNumber) {
+			uiNeedsRendering(&automationInstrumentClipView);
```

Comment:
I think you should look for a cleaner way to do this, this is just going to lead to whack a mole as ways to change a parameter are added. It probably needs to be in the instrument itself

Reply:
> I think you should look for a cleaner way to do this, this is just going to lead to whack a mole as ways to change a parameter are added. It probably needs to be in the instrument itself

This makes sense.

I've moved it to melodic instrument now. Can you take a look if you think this implementation is better?

I don't have a midi controller with me atm so can't test if this works, but it should.